x-pack/plugins/aiops/public/components/explain_log_rate_spikes/field_filter_popover.tsx

```diff
@@ -0,0 +1,221 @@
+  const setFieldsFilter = (fieldNames: string[], checked: boolean) => {
+    let updatedSkippedFields = [...skippedFields];
+    if (!checked) {
+      updatedSkippedFields.push(...fieldNames);
+    } else {
+      updatedSkippedFields = skippedFields.filter((d) => !fieldNames.includes(d));
+    }
+    setSkippedFields(updatedSkippedFields);
+  };
+
+  const [isFieldSelectionPopoverOpen, setIsFieldSelectionPopoverOpen] = useState(false);
+  const onFieldSelectionButtonClick = () => setIsFieldSelectionPopoverOpen((isOpen) => !isOpen);
+  const closePopover = () => setIsFieldSelectionPopoverOpen(false);
+
+  const filteredUniqueFieldNames = useMemo(() => {
+    return uniqueFieldNames.filter(
+      (d) => d.toLowerCase().indexOf(fieldSearchText.toLowerCase()) !== -1
+    );
+  }, [fieldSearchText, uniqueFieldNames]);
+
+  // If the supplied list of unique field names changes, do a sanity check to only
+  // keep field names in the list of skipped fields that still are in the list of unique fields.
+  useEffect(() => {
+    setSkippedFields((previousSkippedFields) =>
+      previousSkippedFields.filter((d) => uniqueFieldNames.includes(d))
+    );
+  }, [uniqueFieldNames]);
+
+  const selectedFieldCount = uniqueFieldNames.length - skippedFields.length;
+
+  return (
+    <EuiPopover
+      data-test-subj="aiopsFieldFilterPopover"
+      anchorPosition="downLeft"
+      panelPaddingSize="s"
+      button={
+        <EuiButton
+          data-test-subj="aiopsFieldFilterButton"
+          onClick={onFieldSelectionButtonClick}
+          disabled={disabled}
+          size="s"
+          iconType="arrowDown"
+          iconSide="right"
+          iconSize="s"
+        >
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterButtonLabel"
+            defaultMessage="Filter fields"
+          />
+        </EuiButton>
+      }
+      isOpen={isFieldSelectionPopoverOpen}
+      closePopover={closePopover}
+    >
+      <EuiPopoverTitle>
+        <EuiText size="xs" color="subdued" style={{ maxWidth: '400px' }}>
+          <FormattedMessage
+            id="xpack.aiops.explainLogRateSpikesPage.fieldFilterHelpText"
+            defaultMessage="To remove non-relevant fields from groups, deselect them below and click the Apply button to rerun the grouping. The search bar can be used to filter the list of fields and is useful to select/deselect in bulk with the actions below the list."
+          />
+        </EuiText>
+        <EuiSpacer size="s" />
+        <EuiFieldText
+          compressed
+          placeholder={i18n.translate('xpack.aiops.analysis.fieldSelectorPlaceholder', {
+            defaultMessage: 'Search',
+          })}
+          aria-label={i18n.translate('xpack.aiops.analysis.fieldSelectorAriaLabel', {
+            defaultMessage: 'Filter fields',
+          })}
+          value={fieldSearchText}
+          onChange={(e: ChangeEvent<HTMLInputElement>) => setFieldSearchText(e.currentTarget.value)}
```

Comment:
Would it be helpful to debounce here?

Reply:
I'll skip doing this for now since there's no immediate remote or async action involved. (The EUI component I took this from works the same way)